Add custom sizes attributes to responsive images

mor10 opened the pull request:
Adds control of `sizes` attribute in responsive images to reflect actual displayed size, not only full viewport width. Fixes #120

mor10 committed:
Adds control of sizes attribute in responsive images to reflect actual displayed size, not only full viewport width. Handles both content images and featured images. Reworked from Twenty Sixteen.

laurelfulford commented on functions.php:
```diff
@@ -255,6 +255,54 @@ function twentyseventeen_scripts() {
+function twentyseventeen_content_image_sizes_attr( $sizes, $size ) {
+	$width = $size[0];
+
+	//if ( 'page' === get_post_type() || ( ! is_active_sidebar( 'sidebar-1' ) && is_single() ) ) {}
```
Getting a minor formatting error here - `No space found before comment text; expected "// if (..."`

It looks like it's commented out code - if that's the case, it can just be removed instead. Thanks!

mor10 committed:
Cleanup. Thanks for catching it @laurelfulford

laurelfulford commented:
Thanks for this PR, @mor10!

I noticed a couple small things I wanted to run by you:

First, I get a bit of a gap for image sizes for the `767px` breakpoint - it's definitely minor, but the `82vw` doesn't seem to cover the whole space all the time:

https://cloudup.com/czv9khWU2q6

What do you think about rolling that one together with the `706px` breakpoint in all cases? Instead of having the two for `706px` and `767px`, it would just be:

`(max-width: 767px) 89vw`

From what I can tell, `89vw` seems to be wide enough to fix the small gap.

Second, when pages are set to use the one column layout in the Customizer, the max image size is 580px:

https://cloudup.com/c-IHFWln_EV

If a check for `'one-column' === get_theme_mod( 'page_options' )` can be incorporated into what you have, with the image size set to `740px` for desktop screen sizes (and smaller for responsive), that should do that trick!

Just let me know if you have any questions at all about the above, and thanks again for this work!

mor10 committed:
Fixes width issue as mentioned by @laurelfulford. Also adds condition for page layouts and sets the right sizes attributes for featured images on index pages.

mor10 commented:
@laurelfulford I've updated the PR with the following:
- Fixed the image width issue you pointed out by simplifying the rules.
- Added conditional for page layouts with a stacked logic chain. I tried a whole bunch of different iterations and this seems like the cleanest one.
- Fixed the sizes attribute for featured images on index pages (it was the wrong no-sidebar size in the original PR).

mor10 committed:
Coding standards

laurelfulford commented on functions.php:
```diff
@@ -255,6 +255,55 @@ function twentyseventeen_scripts() {
+
+	if ( is_active_sidebar( 'sidebar-1' ) || is_archive() || is_search() || is_home() || is_page() ) {
+		if ( ! ( is_page() && 'one-column' === get_theme_mod( 'twentyseventeen_page_options' ) ) ) {
+			767 <= $width && $sizes = '(max-width: 767px) 89vw, (max-width: 1000px) 54vw, (max-width: 1071px) 543px, 580px';
```
The `theme_mods` had their prefixes removed in #23 - `twentyseventeen_page_options` just needs to be switched to `page_options` and this should be good!

laurelfulford commented:
Thanks @mor10! Added a comment for one small issue, but that should be it! 🙂 

Also, to verify - is your WordPress.org username mor10 as well?

mor10 commented:
I'll update the prefix. And yes, my username is mor10 pretty much everywhere.

mor10 committed:
Update get_theme_mod property

laurelfulford commented:
Thanks @mor10!
